doc/cabal-project.rst

```diff
@@ -269,6 +269,36 @@ package, and thus apply globally:
 
     Specifies the name of the directory of the global package store.
 
+.. cfg-field:: package-dbs: package DB stack (comma separated)
+               --package-db=[clear, global, user, PATH]
+    :synopsis: PackageDB stack manipulation.
+    :since: 3.6
+
+    Append the given package database to the list of package databases used to
+    satisfy dependencies.
+
+    The list of package databases form a stack where the top-most (last) db has
+    the highest priority and is also the target for package registration.
+
+    clear
+        Remove all current entries from the stack
+    global
+        Compiler-specific packageDB containing rts, base, etc.
+    user
+        User-specific packageDB. Generally not applicable with v2-* commands.
+    PATH
+        Absolute path to a packageDB.
+    
+    With Nix-style local builds, the default package DB stack is ``[global,
```

Comment:
So if you uses `--package-db`, could not you specify the `<compiler store>` as one of the pieces of the stack? is it included always by default? or you have to set it explicitly with the path?

Reply:
i would change `<compiler store>` with `<cabal store>` as that package db is created and used by cabal. to distinguish it from the global, which is tied to the compiler

Reply:
> Before the change, the PackageDB stack was (relevant code in ProjectPlanning elaboration):
> [CompilerDB, InplaceDB]

> After the change, the PackageDB stack is:
> [CompilerDB, <user flag dbs>, InplaceDB]

If this is still up to date could be used as a model to add the note i miss

Reply:
So I tried to keep this relatively concise, but using my computer (debian) as an example:

`global`: `~/.ghcup/ghc/8.10.7/lib/ghc-8.10.7/package.conf.d`
`user`: not defined
store: `~/.cabal/store/ghc-8.10.7/package.db`
in-place: `./dist-newstyle/packagedb/ghc-8.10.7`

By default, the package stack you will have with v2 commands is:
`[global, store]`
So all remote packages required by your project will be registered in the store package db (because it is last).

When cabal starts building your local projects, it appends the in-place db to the end:
`[global, store, in-place]`
So your local packages get put in dist-newstyle instead of the store.

This flag manipulates the default prefix:
`[global, store]`

To more directly answer your questions:
- yes, you can specify the store package db with: `--package-db /home/patrick/.cabal/store/ghc-8.10.7/package.db`
- yes, it is included by default

The main tricky bit is the `clear` option, which removes all previous entries. Ex.
`--p-db foo` -> `[global, store, foo]`
`--p-db clear --p-db foo` -> `[foo]`
`--p-db clear --p-db foo --p-db clear --p-db bar --p-db baz` -> `[bar, baz]`

Reply:
wow, that is a great explanation, imo it deserves be included in docs including the super useful final examples, would you mind do it in this pr as you habe it almost written?

Reply:
Sure. Just under the flag description or a more general place?

Reply:
in the flag description sounds good to me, it is a somewhat advanced stuff and useful for someone who already wants to tweak seminternal things

Reply:
updated